sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -291,17 +381,49 @@ static az_result connecting(az_event_policy* me, az_event event)
   return ret;
 }
 
-// TODO_L: Implement reconnect logic.
-/*
-static az_result reconnect_timeout(az_event_policy* me, az_event event)
+AZ_INLINE az_result _start_reconnect_timer(az_mqtt5_connection* conn)
 {
+  int32_t random_num = 0;
+  _az_event_pipeline* pipeline = (_az_event_pipeline*)&conn->_internal.event_pipeline;
+  _az_event_pipeline_timer* timer = (_az_event_pipeline_timer*)&conn->_internal.connection_timer;
+
+  _az_FAULTED_IF_FAILED(az_platform_get_random(&random_num), conn, reconnect_timeout);
```

Comment:
I'm a bit nervous about some of the faulted if failed's added in this PR - the more we have the less recoverable any errors would be. As well as - If get_random fails, is that just a problem for this HFSM instance, or should this actually cause platform_panic?

Reply:
All of the errors coming from az_platform now cause critical errors, faulted if failed has been restricted mostly to transitions